### PR TITLE
Refactor of main figures into individual notebooks + name origin of quotes

### PR DESCRIPTION
**Pull request by nrosed**
So again many files changed, but most are figures. Sorry, this PR turned into 2 larger changes instead of one specific change. 

In this PR I did 3 things, 1) small text changes, 2) add an analysis predicting the name origin of quoted speakers, and 3) refactor all my main figures into separate notebooks to make it easier to re-generate the figures.

New Analysis of name origin for quoted speakers:

- `analysis_scripts/analyze_quote_data/name_origin_quotes_with_bg.md`
- `process_doi_data/process_speaker_fullnames.R`

The following files have been reviewed previously but in a different notebook or utils file.
Figure Refactoring:

- `analysis_scripts/analysis_utils.R`
- `utils/plotting_utils.R`
- `figure_notebooks/figure1.md / .Rmd`
- `figure_notebooks/figure2.md / .Rmd`
- `figure_notebooks/figure3.md / .Rmd`
- `figure_notebooks/figure4.md / .Rmd`

### CODE REVIEW

File: figure_notebooks/figure4.md [hunk 1]
```diff
@@ -0,0 +1,997 @@
...
+# now make the heatmap
+res_MC$plot_matr = res_MC$plot_matr[row.names(annot_df),]
+full_heatmap = pheatmap(res_MC$plot_matr, cluster_rows = F, 
+         cluster_cols = F, display_numbers = T, 
+         main = "Top (Mention - Citation) Proportions",
+         color = res_MC$color_pmap, breaks = res_MC$breaks,
+         number_color = "black", annotation_row = annot_df)
+```
+
+![](figure4_files/figure-markdown_github/make_heatmap_gg-1.png)
```
Comment (danich1) [inline figure_notebooks/figure4.md:885]: Heat map looks a bit cramped in this part, but doesn't really matter as it looks great in the main figure. Nice!!

File: analysis_scripts/analyze_quote_data/name_origin_quotes_with_bg.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    ## 20 doi:10.1097/GIM.0b013e3181b13a4f
+    ## 22   doi:10.1186/1743-8454-2-S1-S51
+    ## 23   doi:10.1186/1743-8454-2-S1-S29
+    ## 24   doi:10.1186/1743-8454-2-S1-S28
+
+### QC
+
+From each article type, lets compare the total number of articles considered in this analysis.
+
+<img src="name_origin_quotes_with_bg_files/figure-markdown_github/qc_analysis-1.png" style="display: block; margin: auto;" /><img src="name_origin_quotes_with_bg_files/figure-markdown_github/qc_analysis-2.png" style="display: block; margin: auto;" />
```
Comment (danich1) [inline analysis_scripts/analyze_quote_data/name_origin_quotes_with_bg.md:108]: Any reason for the decreasing number of articles?

**Review by danich1 — APPROVED** [review]
Awesome looking figures in this PR. One comment and question.

File: analysis_scripts/analyze_quote_data/name_origin_quotes_with_bg.md (hunk 2 above)
Comment (nrosed) [reply analysis_scripts/analyze_quote_data/name_origin_quotes_with_bg.md:108]: Yes, unfortunately... Over the years Nature News categories have changed and not all new categories are labeled as time moves forward. One could spider the entire website and then do a lot of post-hoc filtering, but I think it's a bit too much work for the expected benefit.

**Pull request merged**